jep/401/README.adoc

```diff
@@ -0,0 +1,245 @@
+//| Superseded-By
+//| :bulb: JEP-NUMBER :bulb:
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+== Abstract
+
+[TIP]
+====
+Give a short (200 word) description of the technical issue addressed.
+
+* Use present tense - describe what the proposal "does" (as if it were already done), not what it will do.
+* Do not go into technical details and instead put those in the Specification section.
+* Do not talk about history or why this needs to be done. Instead, add the history to the Motivation section.
+====
+The permission types `Jenkins.RUN_SCRIPTS`, `Jenkins.CONFIGURE_UPDATECENTER`, and `Jenkins.UPLOAD_PLUGINS` provide a false sense of security, as a user is able to escalate this limited privilege to `Jenkins.ADMINISTER`.  These permissions are being marked as `@Deprecated` and are now consolidated as `Jenkins.ADMINISTER`.  
+
+A new permission type, `Jenkins.CONFIGURE` (implied by `Jenkins.ADMINISTER`) has been created for actions that allow for the management of Jenkins configuration items that do not require `Jenkins.ADMINISTER`.  This new permission is used to create a clear separation between settings that impact the entirety of a Jenkins instance (particularly with regards to security) and those that can be used to manage other aspects of an installation. 
+
+
+
+== Specification
+
+[TIP]
+====
+Provide a detailed specification of what is being proposed.
+Be as technical and detailed as needed to allow new or existing Jenkins developers
+to reasonably understand the scope/impact of an implementation.
+
+* Use present tense - describe what the proposal "does" (as if it were already done), not what it will do.
+* Do not discuss alternative designs that were rejected - those belong in the Reasoning section.
+* Avoid in-depth discussion or justification of design choices - that belongs in the Reasoning section.
+====
+
+[]
+====
+Angelique - need to add additional info about Jenkins.READ?
+====
+
+The following permission types are now marked as `@Deprecated` and `@Restricted(NoExternalUse.class)`:
+
+* `Jenkins.RUN_SCRIPTS`
+* `Jenkins.CONFIGURE_UPDATECENTER`
+* `Jenkins.UPLOAD_PLUGINS`
+
+This allows for backwards compatiblilty with plugins while providing motivation to refactor any usages of the deprecated checks in future releases.
+
+Actions that formerly checked for these permissions will instead check for `Jenkins.ADMINISTER`.
+
+All other actions that check for `Jenkins.ADMINISTER` will be evaluated to determine if the use of the new `Jenkins.CONFIGURE` permission type is appropriate.  The general rule of thumb is that actions which do not let you change the underlying security model (either directly, or through a malicious plugin or groovy script), or change fundamental configurations such as hostnames, proxy configuration, etc. are likely candidates to use the `Jenkins.CONFIGURE`.
```

Comment:
The rule of thumb is that the configuration being changed should not allow you to escalate to do something that you would not normally be able to do if you just had CONFIGURE.
(I say should because there will be some plugins that have some security issues.)
Thus I think that this needs to be a phased implementation 
1) introduce the new permissions and advertise them (as Preview/Beta?)
2) fixup plugins as bad behaviour is found (probably want to target a subset of the 1000 plugins here!)
3) make the new permission default and deprecate the old ones.

any plugins that make incorrect assumptions should then be treated as security issues.